src/Form/MyTeam/DelegateAccessType.php

```diff
@@ -40,6 +40,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             $accesses[] = DelegatedAccess::ACCESS_COMMITTEE;
         }
 
+        if ('referent' === $options['type']) {
```

Comment:
What about moving this logic into Enum class, for example:
```php
$accesses = DelegatedAccessEnum::getAccessesForType($type);
```

and you could improve allowed `type` values:

```php
public function configureOptions(OptionsResolver $resolver)
{
    $resolver
            ->setDefaults([
                'data_class' => DelegatedAccess::class,
            ])
            ->setDefined('type')
            ->setRequired('type')
            ->setAllowedValues('type', DelegatedAccessEnum::TYPES)
        ;        
}
```